### DISCUSSION TITLE
Changelog of a private GitLab repository readable?

### DISCUSSION THREAD

**Discussion by mrclrchtr** [renovatebot/renovate#8207] [Request Help] [answered]
Is it possible to read the CHANGELOG file of a **private** GitLab repository with a java library (maven)? 
Even better would be to be able to read the release like in GitHub.

We create a tag with release information.Then the pipeline with maven builds the libs in a multi module project and these are published to the gitlab maven repository in the project.

We also maintain a CHANGELOG file in the root of the project.

SCM is defined in the root pom:
```
  <scm>
        <connection>scm:git:https://gitlab.com/xxx/yyy/zzz.git</connection>
        <url>https://gitlab.com/xxx/yyy/zzz</url>
        <developerConnection>scm:git:git@gitlab.com:xxx/yyy/zzz.git</developerConnection>
    </scm>
```

In the projects where the libs are used, PRs are created with the corresponding version updates. However, no release notes are displayed.

(Self Hosted Configuration)

**Comment by rarkins** [comment, marked as answer]
I would have expected it to work if the token you use also has access to the repo in question. Can you maybe create a new test repo with just one dependency and then log it at debug level to see what info it gives us?

**Reply by mrclrchtr** [reply]
Thats exactly what I did today to understand the problem:
[job.txt](https://github.com/renovatebot/renovate/files/5782356/job.txt)
(I have anonymized some parts)

Could it be because I specified the group repository as the repository instead of the repository in the project?

**Reply by rarkins** [reply]
It says the MR is conflicted, so it was not a clean run. Can you do this?

1. Rename and close the existing MR, including deleting the branch
2. Delete your Renovate cache (according to your log it's `data/cache/renovate`)
3. Log a new run after that

**Reply by mrclrchtr** [reply]
Oh sorry! This should be fine now: 
[job.txt](https://github.com/renovatebot/renovate/files/5783137/job.txt)
I have changed some settings, but the release notes are still not there.

**Reply by rarkins** [reply]
When the MR is created, is there a link in the dependency name to the source URL? I'm trying to confirm if that source URL is detected at all to begin with

**Reply by mrclrchtr** [reply]
Yes: https://gitlab.com/XXXX/YYYY/renovatebot-test/-/blob/master/com/XXXX/tool/tool-retrofit-core
It is not correct. It should be https://gitlab.com/XXXX/YYYY/tool/-/blob/master/tool-retrofit-core

it points to the repo in which the PR is created. But it should point to the repo of the tool right?

**Reply by rarkins** [reply]
I normally see just the plain project/repo without deep paths. We could maybe extract in our manager extract function

**Reply by mrclrchtr** [reply]
Okay, is there anything I can do?

For your information: It is about a lib in a multi module project. SCM is only defined in the root project. In the pom.xml of the lib there is no SCM entry. Should I add the entry explicitly in the child project?

**Reply by mrclrchtr** [reply]
Ok, I just tried it and that seems to be the problem. In the child project where SCM is explicitly defined, the release notes are displayed. In the module without SCM not.

**Reply by rarkins** [reply]
In that case I think it's a known issue: https://github.com/renovatebot/renovate/issues/6742

**Reply by mrclrchtr** [reply]
Ok, thanks for your help!
